### PR TITLE
Bug 1869429 - Increase event extra key limit from 15 to 50

### PR DESCRIPTION
**Pull request by travis79**
### Pull Request checklist ###

- [x] **Quality**: This PR builds and tests run cleanly
  - `make test` runs without emitting any warnings
  - `make lint` runs without emitting any errors
- [x] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [x] **Changelog**: This PR includes a changelog entry to `CHANGELOG.md` or an explanation of why it does not need one
  - Any breaking changes to language binding APIs are noted explicitly

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 - Add support for event metric type in server-side JavaScript outputter ([DENG-1736](https://mozilla-hub.atlassian.net/browse/DENG-1736))
 - BREAKING CHANGE: Dropped support for Python 3.7 ([#638](https://github.com/mozilla/glean_parser/pull/638))
 - Add official support for Python 3.11+ ([#638](https://github.com/mozilla/glean_parser/pull/638))
+- events: Increase extra key limit to 50 ([Bug 1869429](https://bugzilla.mozilla.org/show_bug.cgi?id=1869429))
```
Comment (badboy) [inline CHANGELOG.md:13]: This should go under `Unreleased` above.

File: tests/data/schema-violation.yaml [hunk 2]
```diff
@@ -94,3 +94,108 @@ gleantest.event:
...
+      key_48:
+        description: Sample extra key
+        type: string
+      key_49:
+        description: Sample extra key
+        type: string
+      key_50:
+        description: Sample extra key
+        type: string
+      key_51:
```
Comment (badboy) [inline tests/data/schema-violation.yaml:199]: So much sample!

File: tests/test_parser.py [hunk 3]
```diff
@@ -185,6 +185,111 @@ def test_parser_schema_violation():
...
+                type: string
+                key_49:
+                description: Sample extra key
+                type: string
+                key_50:
+                description: Sample extra key
+                type: string
+                key_51:
+                description: Sample extra key
+                type: string
```
Comment (badboy) [inline tests/test_parser.py:292]: At what point should we generate this? :D (all good, keep as is for now)

**Review by badboy — CHANGES_REQUESTED** [review]

File: tests/data/schema-violation.yaml (hunk 2 above)
Comment (travis79) [reply tests/data/schema-violation.yaml:199]: Tell me about it!

File: tests/test_parser.py (hunk 3 above)
Comment (travis79) [reply tests/test_parser.py:292]: The thought did cross my mind when I had to change the second test sample dataset.  🙃

**Review by badboy — APPROVED** [review]

**Pull request merged**